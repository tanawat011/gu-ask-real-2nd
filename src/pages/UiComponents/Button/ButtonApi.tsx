import tw from 'twin.macro'

import { Title } from 'components/PageUiComponent'
import { Table } from 'components/Table'

import { ButtonApiColumns, ButtonApiData } from './data'

const TwContainer = tw.div`mb-8`

export const ButtonApi = () => {
  return (
    <TwContainer>
      <Title title='API' />

      <Table columns={ButtonApiColumns} data={ButtonApiData} />
    </TwContainer>
  )
}
