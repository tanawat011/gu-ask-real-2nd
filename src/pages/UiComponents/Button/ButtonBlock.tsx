import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'

const TwContainer = tw.div`mb-8`

export const ButtonBlock = () => {
  return (
    <TwContainer>
      <Title
        title='Block'
        description={
          <>
            Button can stretch to fill the width by add <code>block</code> prop.
          </>
        }
      />
      <ComponentDisplay
        components={
          <>
            <Button label='Non Block Button' />
            <Button label='Blocked Button' block />
            <Button label='Blocked Button' block loading />
          </>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <>
      <Button label='Non Block Button' />
      <Button label='Blocked Button' block />
      <Button label='Blocked Button' block loading />
    </>
  )
}`}
      />
    </TwContainer>
  )
}
