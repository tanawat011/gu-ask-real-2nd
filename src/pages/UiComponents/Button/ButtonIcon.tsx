import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'

const TwContainer = tw.div`mb-8`

export const ButtonIcon = () => {
  return (
    <TwContainer>
      <Title
        title='Icon'
        description={
          <>
            Button can add an icon by insert icon to <code>icon</code> prop.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button size='xs' icon={<FontAwesomeIcon icon='home' />} />
            <Button size='sm' icon={<FontAwesomeIcon icon='home' />} />
            <Button size='md' icon={<FontAwesomeIcon icon='home' />} />
            <Button size='lg' icon={<FontAwesomeIcon icon='home' />} />
            <Button size='xl' icon={<FontAwesomeIcon icon='home' />} />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button size='xs' icon={<FontAwesomeIcon icon='home' />} />
    <Button size='sm' icon={<FontAwesomeIcon icon='home' />} />
    <Button size='md' icon={<FontAwesomeIcon icon='home' />} />
    <Button size='lg' icon={<FontAwesomeIcon icon='home' />} />
    <Button size='xl' icon={<FontAwesomeIcon icon='home' />} />
  )
}`}
      />
    </TwContainer>
  )
}
