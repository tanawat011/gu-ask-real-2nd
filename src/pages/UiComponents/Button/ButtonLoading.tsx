import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'

const TwContainer = tw.div`mb-8`

export const ButtonLoading = () => {
  return (
    <TwContainer>
      <Title
        title='Loading'
        description={
          <>
            Button can set <code>loading</code> prop to <code>true</code> to show loading indicator,
            it will disabled button.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Home' loading size='xs' />
            <Button label='Home' loading />
            <Button icon={<FontAwesomeIcon icon='home' />} label='Home' loading />
            <Button icon={<FontAwesomeIcon icon='home' />} loading size='xs' />
            <Button icon={<FontAwesomeIcon icon='home' />} loading />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Home' loading size='xs' />
    <Button label='Home' loading />
    <Button icon={<FontAwesomeIcon icon='home' />} label='Home' loading />
    <Button icon={<FontAwesomeIcon icon='home' />} loading size='xs' />
    <Button icon={<FontAwesomeIcon icon='home' />} loading />
  )
}`}
      />
    </TwContainer>
  )
}
