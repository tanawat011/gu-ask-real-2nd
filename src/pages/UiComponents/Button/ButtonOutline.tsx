import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'

const TwContainer = tw.div`mb-8`

export const ButtonOutline = () => {
  return (
    <TwContainer>
      <Title
        title='Outline'
        description={
          <>
            Outline style of the button, can set the <code>outline</code> prop value to{' '}
            <code>true</code> or <code>false</code>, The default outline is <code>false</code>.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Primary' variant='primary' outline />
            <Button label='Secondary' variant='secondary' outline />
            <Button label='Tertiary' variant='tertiary' outline />
            <Button label='Success' variant='success' outline />
            <Button label='Danger' variant='danger' outline />
            <Button label='Warning' variant='warning' outline />
            <Button label='Info' variant='info' outline />
            <Button label='Light' variant='light' outline />
            <Button label='Dark' variant='dark' outline />
            <Button label='Link' variant='link' outline />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Primary' variant='primary' outline />
    <Button label='Secondary' variant='secondary' outline />
    <Button label='Tertiary' variant='tertiary' outline />
    <Button label='Success' variant='success' outline />
    <Button label='Danger' variant='danger' outline />
    <Button label='Warning' variant='warning' outline />
    <Button label='Info' variant='info' outline />
    <Button label='Light' variant='light' outline />
    <Button label='Dark' variant='dark' outline />
    <Button label='Link' variant='link' outline />
  )
}`}
      />
    </TwContainer>
  )
}
