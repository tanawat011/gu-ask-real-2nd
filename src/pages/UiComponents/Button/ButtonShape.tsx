import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'

const TwContainer = tw.div`mb-8`

export const ButtonShape = () => {
  return (
    <TwContainer>
      <Title
        title='Shape'
        description={
          <>
            Shape style of the button, can set the <code>shape</code> prop value to{' '}
            <code>square</code>, <code>rounded</code> or <code>circle</code>, The default shape is{' '}
            <code>rounded</code>.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Square' shape='square' />
            <Button label='Rounded' shape='rounded' />
            <Button label='Circle' shape='circle' />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Square' shape='square' />
    <Button label='Rounded' shape='rounded' />
    <Button label='Circle' shape='circle' />
  )
}`}
      />
    </TwContainer>
  )
}
