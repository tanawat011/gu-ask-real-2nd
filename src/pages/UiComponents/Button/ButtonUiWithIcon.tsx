import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'

const TwContainer = tw.div`mb-8`

export const ButtonUiWithIcon = () => {
  return (
    <TwContainer>
      <Title
        title='With Icon'
        description={
          <>
            Icon can add and combined with <code>label</code>.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button icon={<FontAwesomeIcon icon='home' />} label='Home' />
            <Button icon={<FontAwesomeIcon icon='user' />} label='Profile' />
            <Button icon={<FontAwesomeIcon icon='gear' />} label='Setting' />
            <Button iconR={<FontAwesomeIcon icon='home' />} label='Right Icon' />
            <Button
              icon={<FontAwesomeIcon icon='home' />}
              iconR={<FontAwesomeIcon icon='home' />}
              label='Between Icons'
            />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button icon={<FontAwesomeIcon icon='home' />} label='Home' />
    <Button icon={<FontAwesomeIcon icon='user' />} label='Profile' />
    <Button icon={<FontAwesomeIcon icon='gear' />} label='Setting' />
    <Button iconR={<FontAwesomeIcon icon='home' />} label='Right Icon' />
    <Button
      icon={<FontAwesomeIcon icon='home' />}
      iconR={<FontAwesomeIcon icon='home' />}
      label='Between Icons'
    />
  )
}`}
      />
    </TwContainer>
  )
}
