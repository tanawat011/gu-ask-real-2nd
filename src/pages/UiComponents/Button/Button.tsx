import tw from 'twin.macro'

import { Title } from 'components/PageUiComponent'

import { ButtonBlock } from './ButtonBlock'
import { ButtonDisabled } from './ButtonDisabled'
import { ButtonIcon } from './ButtonIcon'
import { ButtonLoading } from './ButtonLoading'
import { ButtonOutline } from './ButtonOutline'
import { ButtonShape } from './ButtonShape'
import { ButtonSize } from './ButtonSize'
import { ButtonVariant } from './ButtonVariant'
import { ButtonWithIcon } from './ButtonWithIcon'
import { RightMenuButton } from './RightMenuButton'

const TwContainer = tw.div`grid grid-cols-1 gap-8 px-8 py-6 desktop:grid-cols-5`
const TwWrapContent = tw.div`xl:col-span-4`
const TwWrapMenu = tw.div`col-span-1 hidden desktop:block`

export const ButtonUi = () => {
  return (
    <>
      <TwContainer>
        <TwWrapContent>
          <Title
            size='xl'
            title='Button'
            description='Button is used to triggers an action or event click'
          />

          <ButtonVariant />
          <ButtonSize />
          <ButtonOutline />
          <ButtonShape />
          <ButtonDisabled />
          <ButtonIcon />
          <ButtonWithIcon />
          <ButtonLoading />
          <ButtonBlock />
        </TwWrapContent>

        <TwWrapMenu>
          <RightMenuButton />
        </TwWrapMenu>
      </TwContainer>
    </>
  )
}
