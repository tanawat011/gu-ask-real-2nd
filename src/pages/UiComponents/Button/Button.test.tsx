import type { Children } from 'types'

import { mocked } from 'jest-mock'

import { useIsScrolled } from 'hooks/useIsScrolled'
import { renderWithProviders } from 'utils/jest'

import { ButtonUi } from './Button'

jest.mock('@fortawesome/react-fontawesome')
jest.mock('components/PageUiComponent', () => ({
  ComponentDisplay: ({ children }: Children) => <div>{children}</div>,
  RightMenuContent: ({ children }: Children) => <div>{children}</div>,
  Title: () => <div>Title</div>,
}))
jest.mock('components/Table', () => ({
  Table: () => (
    <table>
      <thead>
        <tr>
          <th>1</th>
          <th>2</th>
          <th>3</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>x</td>
          <td>y</td>
          <td>z</td>
        </tr>
      </tbody>
    </table>
  ),
}))
jest.mock('components/Button', () => ({
  Button: () => <button>Button</button>,
}))
jest.mock('hooks/useIsScrolled')

describe('PageUiComponent <ButtonUi />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = renderWithProviders(<ButtonUi />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with scrolled', () => {
    expect.assertions(1)

    mocked(useIsScrolled).mockReturnValue(true)

    const { asFragment } = renderWithProviders(<ButtonUi />)

    expect(asFragment()).toMatchSnapshot()
  })
})
