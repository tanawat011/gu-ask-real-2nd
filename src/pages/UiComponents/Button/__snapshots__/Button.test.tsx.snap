// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageUiComponent <ButtonUi /> renders correctly 1`] = `
<DocumentFragment>
  .c11 {
  margin-bottom: 2rem;
}

.c10 {
  margin-bottom: 2rem;
}

.c6 {
  margin-bottom: 2rem;
}

.c7 {
  margin-bottom: 2rem;
}

.c9 {
  margin-bottom: 2rem;
}

.c4 {
  margin-bottom: 2rem;
}

.c5 {
  margin-bottom: 2rem;
}

.c3 {
  margin-bottom: 2rem;
}

.c2 {
  margin-bottom: 2rem;
}

.c8 {
  margin-bottom: 2rem;
}

.c13 {
  z-index: 10;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(1,minmax(0,1fr));
  gap: 2rem;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.c12 {
  grid-column: span 1 / span 1;
  display: none;
}

@media (min-width:1280px) {
  .c0 {
    grid-template-columns: repeat(5,minmax(0,1fr));
  }
}

@media (min-width:1280px) {
  .c1 {
    grid-column: span 4 / span 4;
  }
}

@media (min-width:1280px) {
  .c12 {
    display: block;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        Title
      </div>
      <div
        class="c2"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c3"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c4"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c5"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c6"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c7"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c8"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c9"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c10"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c11"
      >
        <div>
          Title
        </div>
        <table>
          <thead>
            <tr>
              <th>
                1
              </th>
              <th>
                2
              </th>
              <th>
                3
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                x
              </td>
              <td>
                y
              </td>
              <td>
                z
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c13"
      >
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PageUiComponent <ButtonUi /> renders with scrolled 1`] = `
<DocumentFragment>
  .c11 {
  margin-bottom: 2rem;
}

.c10 {
  margin-bottom: 2rem;
}

.c6 {
  margin-bottom: 2rem;
}

.c7 {
  margin-bottom: 2rem;
}

.c9 {
  margin-bottom: 2rem;
}

.c4 {
  margin-bottom: 2rem;
}

.c5 {
  margin-bottom: 2rem;
}

.c3 {
  margin-bottom: 2rem;
}

.c2 {
  margin-bottom: 2rem;
}

.c8 {
  margin-bottom: 2rem;
}

.c13 {
  z-index: 10;
  position: fixed;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(1,minmax(0,1fr));
  gap: 2rem;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.c12 {
  grid-column: span 1 / span 1;
  display: none;
}

@media (min-width:1280px) {
  .c0 {
    grid-template-columns: repeat(5,minmax(0,1fr));
  }
}

@media (min-width:1280px) {
  .c1 {
    grid-column: span 4 / span 4;
  }
}

@media (min-width:1280px) {
  .c12 {
    display: block;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        Title
      </div>
      <div
        class="c2"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c3"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c4"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c5"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c6"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c7"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c8"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c9"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c10"
      >
        <div>
          Title
        </div>
        <div />
      </div>
      <div
        class="c11"
      >
        <div>
          Title
        </div>
        <table>
          <thead>
            <tr>
              <th>
                1
              </th>
              <th>
                2
              </th>
              <th>
                3
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                x
              </td>
              <td>
                y
              </td>
              <td>
                z
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c13"
      >
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
